File: frontend/src/app/features/credit/models/demande-credit-response.ts
import { DureeUnite } from "../../../shared/enums/duree-unite.enum";
import { PeriodiciteRemboursement } from "../../../shared/enums/periodicite-remboursement.enum";
import { StatutDemandeCredit } from "./enums/statut-demande-credit.enum";

export interface DemandeCreditResponse {
  id: number;
  numeroDemande: string;

  membreId: number;
  membreNomComplet: string;

  siteId: number;
  siteNom: string;
  agentId?: number | null;

  dateDemande: string;

  montantDemande: number;

  fraisDemande: number;
  fraisDemandePayes: number;

  depotGarantieRequis: number;
  depotGarantiePaye: number;
  montantGarantieBloque?: number;
  garantieBloquee?: boolean;
  statutGarantie?: 'NON_BLOQUEE' | 'BLOQUEE' | 'INSUFFISANTE';

  devise: string;

  dureeValeur: number;
  dureeUnite: DureeUnite;
  periodiciteRemboursement: PeriodiciteRemboursement;

  tauxInteret: number;

  objetCredit: string;
  gagePropose?: string | null;
  activiteFinancee?: string | null;

  revenusEstimes: number;
  chargesEstimees: number;

  statut: StatutDemandeCredit;

  commentaireDecision?: string | null;
  dateDecision?: string | null;

  createdAt?: string | null;
  updatedAt?: string | null;
  analyseRisque?: AnalyseRisqueCreditResponse | null;
}

export interface AnalyseRisqueCreditResponse {
  scoreTotal: number;
  niveauRisque: 'FAIBLE' | 'MOYEN' | 'ELEVE';
  montantDemande: number;
  revenusMensuels: number;
  chargesMensuelles: number;
  capaciteRemboursement: number;
  mensualiteEstimee: number;
  ratioMensualiteCapacite?: number | null;
  ratioMontantRevenu?: number | null;
  garantieEpargneRequise: number;
  garantieEpargneDisponible: number;
  garantieMaterielleRequise?: number | null;
  garantieMaterielleDeclaree?: number | null;
  criteres: CritereAnalyseRisque[];
}

export interface CritereAnalyseRisque {
  codeCritere: string;
  libelle: string;
  points: number;
  niveau: string;
  commentaire: string;
}

File: backend/src/main/java/com/mini/credit/dto/credit/DemandeCreditResponse.java
package com.mini.credit.dto.credit;

import com.mini.credit.enums.DureeUnite;
import com.mini.credit.enums.PeriodiciteRemboursement;
import com.mini.credit.enums.StatutDemandeCredit;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Builder
public class DemandeCreditResponse {
    private Long id;
    private String numeroDemande;
    private Long membreId;
    private String membreNomComplet;
    private Long siteId;
    private String siteNom;
    private Long agentId;
    private LocalDate dateDemande;
    private BigDecimal montantDemande;
    private BigDecimal fraisDemandePayes;
    private String devise;
    private Integer dureeValeur;
    private DureeUnite dureeUnite;
    private PeriodiciteRemboursement periodiciteRemboursement;
    private BigDecimal tauxInteret;
    private String objetCredit;
    private String gagePropose;
    private String activiteFinancee;
    private BigDecimal revenusEstimes;
    private BigDecimal chargesEstimees;
    private BigDecimal fraisDemande;
    private BigDecimal depotGarantieRequis;
    private BigDecimal depotGarantiePaye;
    private BigDecimal montantGarantieBloque;
    private Boolean garantieBloquee;
    private String statutGarantie;
    private StatutDemandeCredit statut;
    private String commentaireDecision;
    private LocalDateTime dateDecision;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private AnalyseRisqueCreditResponse analyseRisque;
}